worker/crates/action-sink/src/file/cesium3dtiles.rs

```diff
@@ -0,0 +1,380 @@
+    geometric_error, x_slice_range, x_step, y_slice_range, zxy_from_lng_lat,
+};
+use reearth_flow_common::uri::Uri;
+use reearth_flow_runtime::errors::BoxedError;
+use reearth_flow_runtime::event::EventHub;
+use reearth_flow_runtime::executor_operation::{ExecutorContext, NodeContext};
+use reearth_flow_runtime::node::{Port, Sink, SinkFactory, DEFAULT_PORT};
+use reearth_flow_storage::resolve::StorageResolver;
+use reearth_flow_types::geometry as geomotry_types;
+use reearth_flow_types::Expr;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use std::collections::HashMap;
+use std::fs;
+use std::path::Path;
+use std::str::FromStr;
+use std::sync::Arc;
+use std::sync::Mutex;
+use std::vec;
+
+#[derive(Debug, Clone, Default)]
+pub struct Cesium3DTilesSinkFactory;
+
+impl SinkFactory for Cesium3DTilesSinkFactory {
+    fn name(&self) -> &str {
+        "Cesium3DTilesWriter"
+    }
+
+    fn description(&self) -> &str {
+        "Writes features to a file"
+    }
+
+    fn parameter_schema(&self) -> Option<schemars::schema::RootSchema> {
+        Some(schemars::schema_for!(Cesium3dtilesWriterParam))
+    }
+
+    fn categories(&self) -> &[&'static str] {
+        &["File"]
+    }
+
+    fn get_input_ports(&self) -> Vec<Port> {
+        vec![DEFAULT_PORT.clone()]
+    }
+
+    fn prepare(&self) -> Result<(), BoxedError> {
+        Ok(())
+    }
+
+    fn build(
+        &self,
+        _ctx: NodeContext,
+        _event_hub: EventHub,
+        _action: String,
+        with: Option<HashMap<String, Value>>,
+    ) -> Result<Box<dyn Sink>, BoxedError> {
+        let params = if let Some(with) = with {
+            let value: Value = serde_json::to_value(with).map_err(|e| {
+                SinkError::BuildFactory(format!("Failed to serialize `with` parameter: {}", e))
+            })?;
+            serde_json::from_value(value).map_err(|e| {
+                SinkError::BuildFactory(format!("Failed to deserialize `with` parameter: {}", e))
+            })?
+        } else {
+            return Err(
+                SinkError::BuildFactory("Missing required parameter `with`".to_string()).into(),
+            );
+        };
+
+        let sink = Cesium3dtilesWriter {
+            params,
+            contents: Arc::new(Mutex::new(Vec::new())),
+        };
+        Ok(Box::new(sink))
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct Cesium3dtilesWriter {
+    pub(super) params: Cesium3dtilesWriterParam,
+    pub(super) contents: Arc<Mutex<Vec<TileContent>>>,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone, JsonSchema)]
+#[serde(rename_all = "camelCase")]
+pub struct Cesium3dtilesWriterCommonParam {
+    pub(super) output: Expr,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone, JsonSchema)]
+#[serde(rename_all = "camelCase", tag = "format")]
+pub enum Cesium3dtilesWriterParam {
+    Cesium3dtiles {
+        #[serde(flatten)]
+        common_property: Cesium3dtilesWriterCommonParam,
+    },
+}
```

Comment:
```suggestion
#[derive(Serialize, Deserialize, Debug, Clone, JsonSchema)]
#[serde(rename_all = "camelCase")]
pub struct Cesium3dtilesWriterParam {
   pub(super) output: Expr,
}
```
In this case, since it is not per-format, you can use “struct” instead of “enum”.

Reply:
Changed it to a struct